### ISSUE TITLE
"Uncaught TypeError: Cannot read property 'nTr' of undefined" after calling clear() and then .row(selector).node() with a jQuery object.

### ISSUE THREAD

**Issue by ioev** [DataTables/DataTables#994]
Issue can be seen here:
http://jsfiddle.net/q32awwjp/3/

After calling .clear() on the table, and then trying to select a row based on a row that used to be in the table, the response returned makes it appear that the row is still present, and additional row methods will fail.

You can see the call using a string selector works as expected, no row is returned and node does not fail.  But calling .node() on the row that is returned from the jquery object fails.  Other row object methods (such as .remove()) will also fail in different ways.

**Comment by ioev** [comment]
Just a bit more info, there seems to be some weird behavior with remove as well, using a jquery object.
```
datatable.row(jquery_object).remove()
datatable.draw(true)
```

Will remove the row as expected, but a call to:
`datatable.row(jquery_object).node()`

still returns the row, as if it still exists in the table.

`datatable.row(id_of_that_row).node()`

returns null as expected.

**Comment by DataTables** [comment]
Thanks for the test case - I've committed a [fix here](https://github.com/DataTables/DataTablesSrc/commit/0348919e615164122c877ef54309fd2116436ee1) and the nightly will rebuild shortly.

There can still be slightly "weird" effects when using cached values as selectors after deleting rows - specifically integers (i.e. row indexes). Those are not updated, and that is a know issue, but this addresses specifically using nodes which I believe is now fixed.

The plan is to use a sparse array in future to address in index issue as well.
